scripts/validate_documentation.py

```diff
@@ -0,0 +1,227 @@
+import argparse
+from fnmatch import fnmatch
+import json
+import os
+import pickle
+import re
+import sys
+
+import docutils.nodes
+
+DOCUMENTATION_SOURCE = os.path.join(os.curdir, '../doc/source')
+DOCTREE_PATH = '../doc/build/doctrees/{}.doctree'
+RST_PATH = '../doc/source/{}.rst'
```

Comment:
I think you are assuming here that the current directory (`pwd`) is `scripts/`, but the script can also be called as `./scripts/validate_documentation.py`, and I think those paths will be wrong. It's usually a good practice to have a `BASE_PATH` first that is the root of the project, and use it as the base for all those. I'd say you should find the code in other scripts.

Also, better use `os.path.join` instead of the hardcoded `/`, so this can also run in windows.